ui/src/core/xibo-calendar.js

```diff
@@ -40,6 +40,133 @@ $(document).ready(function() {
         });
     });
 
+    // Set a listener for type change event
+    $('body').on('change', 'select[name="criteria_type[]"]', function() {
+        // Get the row where the type was changed
+        var $row = $(this).closest('.form-group');
+
+        var selectedType = $(this).val();
+        var $fields = $('#scheduleCriteriaFields');
+        var scheduleCriteria = $fields.data('scheduleCriteria');
+
+        if (scheduleCriteria) {
+            if (selectedType === 'custom') {
+                // Use a text input for metrics
+                updateMetricsFieldAsText($row);
+
+                // Use a text input for values
+                updateValueFieldAsText($row);
+            } else if (scheduleCriteria) {
+
```

Comment:
This is a minor suggestion, since you'll probably update this file, it would be good to remove unnecessary padding from blocks :) (https://eslint.org/docs/latest/rules/padded-blocks)